org.eclipse.jdt.ui.tests/ui/org/eclipse/jdt/junit/launcher/AdvancedJUnitLaunchConfigurationDelegateTest.java

```diff
@@ -192,9 +193,21 @@ private List<String> showCommandLineAndExtractContentOfTestNameFile(String proje
 		String firstSearchStr= "-testNameFile";
 		int indexTestNameFile= showCommandLine.indexOf(firstSearchStr);
 		assertThat(indexTestNameFile).overridingErrorMessage("-testNameFile argument not found").isGreaterThan(-1);
-		String filePath= showCommandLine.substring(indexTestNameFile + firstSearchStr.length() + 1);
+		String filePath= extractPathForArgumentFile(showCommandLine, firstSearchStr, indexTestNameFile);
 		List<String> fileLines= Files.readAllLines(Paths.get(filePath));
 		return fileLines;
 	}
 
+	private String extractPathForArgumentFile(String showCommandLine, String firstSearchStr, int indexTestNameFile) {
+		String filePath= showCommandLine.substring(indexTestNameFile + firstSearchStr.length() + 1);
+		if (Platform.OS_WIN32.equals(Platform.getOS())) {
```

Comment:
I would prefer to have the Test OS independent. Can you always - on all OS - remove the quotation marks? they are not part of the filename i guess.

Reply:
I took another look. The quotation marks are only added on Windows, in `CommandLineQuoting.quoteWindowsArgs`. It is reached from the test as follows:
* JUnitLaunchConfigurationDelegate.showCommandLine
* org.eclipse.jdt.internal.launching.StandardVMDebugger.showCommandLine
* org.eclipse.jdt.internal.launching.StandardVMDebugger.quoteWindowsArgs
* org.eclipse.jdt.internal.launching.CommandLineQuoting.quoteWindowsArgs

I don't see a way to make this test OS-independent when parsing the command line.

I did evaluate whether it was possible to get at the argument file or its contents without parsing the command line when I wrote the original pull request. IIRC I did not find a way to do that without doing large changes to the code.

Reply:
since the impl is OS dependend, the test is OK to have the same dependency